geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/sql/GeoMesaSparkSql.scala

```diff
@@ -0,0 +1,345 @@
+    }
+
+    // combine the two maps
+    namesFromSql.toMap ++ namesFromCql.map { case (k,v) =>
+      k -> namesFromSql.get(k).map(_ ++ v.toSet).getOrElse(v.toSet)
+    }
+  }
+
+  /**
+   * Converts a simple feature attribute into a SQL data type
+   */
+  private def types(d: AttributeDescriptor): DataType = {
+    val clas = d.getType.getBinding
+    if (classOf[jList[_]].isAssignableFrom(clas)) {
+      val listClass = d.getUserData.get(USER_DATA_LIST_TYPE).asInstanceOf[Class[_]]
+      DataTypes.createArrayType(types(listClass))
+    } else if (classOf[jMap[_, _]].isAssignableFrom(clas)) {
+      val keyClass   = d.getUserData.get(USER_DATA_MAP_KEY_TYPE).asInstanceOf[Class[_]]
+      val valueClass = d.getUserData.get(USER_DATA_MAP_VALUE_TYPE).asInstanceOf[Class[_]]
+      DataTypes.createMapType(types(keyClass), types(valueClass))
+    } else {
+      types(clas)
+    }
+  }
+
+  /**
+   * Converts a simple class type into a SQL data type
+   */
+  private def types(clas: Class[_]): DataType = {
+    if (classOf[java.lang.String].isAssignableFrom(clas)) {
+      StringType
+    } else if (classOf[java.lang.Integer].isAssignableFrom(clas)) {
+      IntegerType
+    } else if (classOf[java.lang.Long].isAssignableFrom(clas)) {
+      LongType
+    } else if (classOf[java.lang.Float].isAssignableFrom(clas)) {
+      FloatType
+    } else if (classOf[java.lang.Double].isAssignableFrom(clas)) {
+      DoubleType
+    } else if (classOf[java.lang.Boolean].isAssignableFrom(clas)) {
+      BooleanType
+    } else if (classOf[java.util.Date].isAssignableFrom(clas)) {
+      TimestampType
+    } else if (classOf[UUID].isAssignableFrom(clas)) {
+      StringType
+    } else if (classOf[Geometry].isAssignableFrom(clas)) {
+      StringType
+    } else {
+      throw new NotImplementedError(s"Binding $clas is not supported")
+    }
+  }
+}
+
+class GeoMesaSparkSql(sc: SparkContext, dsParams: Seq[Map[String, String]]) {
+
+  // load up our sfts
+  val sftsByName = dsParams.flatMap { params =>
+    val ds = DataStoreFinder.getDataStore(params)
+    require(ds != null, "No data store found using provided parameters")
+    ds.getTypeNames.map { name =>
+      val schema = ds.getSchema(name)
+      name -> (schema,  params)
+    }
+  }.foldLeft(Map.empty[String, (SimpleFeatureType, Map[String, String])])(_ + _)
+
+  private val dataContext = new GeoMesaDataContext(sftsByName.mapValues(_._1))
+
+  /**
+   * Execute a sql query against geomesa. Where clause is interpreted as CQL.
+   */
+  def query(sql: String, splits: Option[Int]): DataFrame = {
+    val parsedSql = dataContext.parseQuery(sql)
+
+    // extract the feature types from the from clause
+    val typeNames = parsedSql.getFromClause.getItems.map(_.getTable.getName)
+    val sftsWithParams = typeNames.map(sftsByName.apply)
+
+    // extract the cql from the where clause
+    val where = parsedSql.getWhereClause
+    val cql = GeoMesaSparkSql.extractCql(where, dataContext, typeNames)
```

Comment:
There are some sample requests in the readme, but basically you just add it to the where clause like so:
select mySft.myAttr, count(*) as count from mySft where bbox(mySft.geom, -115, 45, -110, 50) AND mySft.dtg during 2015-03-02T10:00:00.000Z/2015-03-02T11:00:00.000Z group by myattr
When we parse out the SQL, if it's not recognized as a normal SQL statement, it doesn't get parsed into an object - so we detect those portions and treat them as CQL.  I'm sure it's not foolproof, but it works for basic cases. In particular, attribute queries will probably end up run by the SQL engine instead of the initial geomesa query.